src/singletons/ImageUploader.cpp

```diff
@@ -259,110 +259,118 @@ void ImageUploader::upload(const QMimeData *source, ChannelPtr channel,
-    if (source->hasUrls())
-    {
-        auto mimeDb = QMimeDatabase();
-        // This path gets chosen when files are copied from a file manager, like explorer.exe, caja.
-        // Each entry in source->urls() is a QUrl pointing to a file that was copied.
-        for (const QUrl &path : source->urls())
+    auto tryUploadFromUrls = [&]() -> bool {
+        if (source->hasUrls())
         {
-            QString localPath = path.toLocalFile();
-            QMimeType mime = mimeDb.mimeTypeForUrl(path);
-            if (mime.name().startsWith("image") && !mime.inherits("image/gif"))
+            auto mimeDb = QMimeDatabase();
+            // This path gets chosen when files are copied from a file manager, like explorer.exe, caja.
+            // Each entry in source->urls() is a QUrl pointing to a file that was copied.
+            for (const QUrl &path : source->urls())
             {
-                channel->addMessage(makeSystemMessage(
-                    QString("Uploading image: %1").arg(localPath)));
-                QImage img = QImage(localPath);
-                if (img.isNull())
+                QString localPath = path.toLocalFile();
+                QMimeType mime = mimeDb.mimeTypeForUrl(path);
+                if (mime.name().startsWith("image") &&
+                    !mime.inherits("image/gif"))
                 {
-                    channel->addMessage(
-                        makeSystemMessage(QString("Couldn't load image :(")));
-                    this->uploadMutex_.unlock();
-                    return;
-                }
+                    channel->addMessage(makeSystemMessage(
+                        QString("Uploading image: %1").arg(localPath)));
+                    QImage img = QImage(localPath);
+                    if (img.isNull())
+                    {
+                        channel->addMessage(makeSystemMessage(
+                            QString("Couldn't load image :(")));
+                        return false;
+                    }
 
-                auto imageData = convertToPng(img);
-                if (imageData)
-                {
-                    RawImageData data = {*imageData, "png", localPath};
-                    this->uploadQueue_.push(data);
+                    auto imageData = convertToPng(img);
+                    if (imageData)
+                    {
+                        RawImageData data = {*imageData, "png", localPath};
+                        this->uploadQueue_.push(data);
+                    }
+                    else
+                    {
+                        channel->addMessage(makeSystemMessage(
+                            QString("Cannot upload file: %1. Couldn't convert "
+                                    "image to png.")
+                                .arg(localPath)));
+                        return false;
+                    }
                 }
-                else
+                else if (mime.inherits("image/gif"))
```

Comment:
To me the else immediately signifies two different code paths that are not mingling, without having to reason about the contents of the if condition and/or the statements.

I think it makes it clearer, especially since the if parts are of somewhat larger size of the content.